ambry-router/src/test/java/com.github.ambry.router/RouterTestHelpers.java

```diff
@@ -140,11 +140,22 @@ static void setServerErrorCodes(ServerErrorCode[] serverErrorCodesInOrder, Parti
     if (serverErrorCodesInOrder.length > replicas.size()) {
       throw new IllegalArgumentException("More server error codes provided than replicas in partition");
     }
+    // Do serverErrorCodes assignment based on Operation Tracker logic so that server error can return in the order of
+    // serverErrorCodesInOrder.
+    int numOfNodesEachDatacenter = 3;
+    ServerErrorCode[] serverErrorCodesInOperationTrackerOrder = new ServerErrorCode[serverErrorCodesInOrder.length];
+    for (int i = 0; i < numOfNodesEachDatacenter; i++) {
+      serverErrorCodesInOperationTrackerOrder[i] = serverErrorCodesInOrder[numOfNodesEachDatacenter - 1 - i];
+    }
+    for (int i = numOfNodesEachDatacenter; i < serverErrorCodesInOrder.length; i++) {
+      serverErrorCodesInOperationTrackerOrder[i] =
+          serverErrorCodesInOrder[serverErrorCodesInOrder.length + numOfNodesEachDatacenter - 1 - i];
+    }
```

Comment:
I am thinking if we can replace this piece of code with a more intuitive one. How about this?
`    List<ServerErrorCode> errorCodesInAllDCs = Arrays.asList(serverErrorCodesInOrder);
    List<ServerErrorCode> serverErrorInLocalDC = errorCodesInAllDCs.subList(0, 3);
    List<ServerErrorCode> serverErrorInOtherDCs = errorCodesInAllDCs.subList(3, serverErrorCodesInOrder.length);
    Collections.reverse(serverErrorInLocalDC);
    Collections.reverse(serverErrorInOtherDCs);
    List<ServerErrorCode> serverErrorCodesInOperationTrackerOrder = new ArrayList<>(serverErrorInLocalDC);
    serverErrorCodesInOperationTrackerOrder.addAll(serverErrorInOtherDCs);`

Reply:
good suggestion!